File: pizzaOrder.py
class Topping:
    def __init__(self, name, price):
        self.name = name
        self.price = price

class Pizza:
    def __init__(self):
        self.base_price = 15.00
        self.toppings = []

    def add_topping(self, topping):
        self.toppings.append(topping)

    def get_price(self):
        total_price = self.base_price
        for topping in self.toppings:
            total_price += topping.price
        return total_price

class Order:
    def __init__(self):
        self.pizzas = []
        
    def add_pizza(self, pizza):
        self.pizzas.append(pizza)

    def get_total_price(self):
        total_sum = 0
        for pizza in self.pizzas:
            total_sum += pizza.get_price()
        return total_sum


menu = [
    Topping("Ser", 3.00),
    Topping("Pepperoni", 3.50),
    Topping("Pieczarki", 2.00),
    Topping("Oliwki", 2.50)
]

print("Witaj w wirtualnej pizzerii! Zaczynamy przygotowywać Twoją pizzę.")
moja_pizza = Pizza()

while True:
    print("\n--- MENU DODATKÓW ---")
    for indeks, dodatek in enumerate(menu):
        print(f"{indeks + 1}. {dodatek.name} ({dodatek.price} zł)")
    print("0. Zakończ dodawanie i podlicz rachunek")
    
    wybor = input("\nWybierz numer opcji: ")
    
    if wybor == '0':
        break 
    elif wybor in ['1', '2', '3', '4']:
        wybrany_indeks = int(wybor) - 1 
        wybrany_dodatek = menu[wybrany_indeks]
        
        moja_pizza.add_topping(wybrany_dodatek)
        print(f"Dodano: {wybrany_dodatek.name}!")
    else:
        print("Nieznana opcja. Spróbuj ponownie.")

moje_zamowienie = Order()
moje_zamowienie.add_pizza(moja_pizza)

print(f"\n Gotowe! Cena całkowita Twojego zamówienia to: {moje_zamowienie.get_total_price()} zł")

File: pizzaOrder.cs
using System;
using System.Collections.Generic;

public class Topping
{
    public string Name;
    public double Price;

    public Topping(string name, double price)
    {
        this.Name = name;
        this.Price = price;
    }
}

public class Pizza
{
    public double BasePrice;
    public List<Topping> Toppings;

    public Pizza()
    {
        this.BasePrice = 15.00;
        this.Toppings = new List<Topping>();
    }

    public void AddTopping(Topping topping)
    {
        this.Toppings.Add(topping);
    }

    public double GetPrice()
    {
        double totalPrice = this.BasePrice;
        foreach (Topping topping in this.Toppings)
        {
            totalPrice += topping.Price;
        }
        return totalPrice;
    }
}

public class Order
{
    public List<Pizza> Pizzas;

    public Order()
    {
        this.Pizzas = new List<Pizza>();
    }

    public void AddPizza(Pizza pizza)
    {
        this.Pizzas.Add(pizza);
    }

    public double GetTotalPrice()
    {
        double totalSum = 0;
        foreach (Pizza pizza in this.Pizzas)
        {
            totalSum += pizza.GetPrice();
        }
        return totalSum;
    }
}

public class Program
{
    public static void Main()
    {
        List<Topping> menu = new List<Topping>
        {
            new Topping("Ser", 3.00),
            new Topping("Pepperoni", 3.50),
            new Topping("Pieczarki", 2.00),
            new Topping("Oliwki", 2.50)
        };

        Console.WriteLine("Witaj w wirtualnej pizzerii! Zaczynamy przygotowywać Twoją pizzę.");
        Pizza mojaPizza = new Pizza();

        while (true)
        {
            Console.WriteLine("\n--- MENU DODATKÓW ---");
            for (int i = 0; i < menu.Count; i++)
            {
                Console.WriteLine($"{i + 1}. {menu[i].Name} ({menu[i].Price} zł)");
            }
            Console.WriteLine("0. Zakończ dodawanie i podlicz rachunek");

            Console.Write("\nWybierz numer opcji: ");
            string wybor = Console.ReadLine();

            if (wybor == "0")
            {
                break;
            }
            
            if (int.TryParse(wybor, out int numer) && numer >= 1 && numer <= menu.Count)
            {
                Topping wybranyDodatek = menu[numer - 1];
                mojaPizza.AddTopping(wybranyDodatek);
                Console.WriteLine($"Dodano: {wybranyDodatek.Name}!");
            }
            else
            {
                Console.WriteLine("Nieznana opcja. Spróbuj ponownie.");
            }
        }

        Order mojeZamowienie = new Order();
        mojeZamowienie.AddPizza(mojaPizza);

        Console.WriteLine($"\nGotowe! Cena całkowita Twojego zamówienia to: {mojeZamowienie.GetTotalPrice()} zł");
    }
}